### PR TITLE
machine/atsamd21: refactor SPI pin handling to only look at pin numbers

### PR DESCRIPTION
**Pull request by aykevl**
Pin mode and pad numbers are automatically calculated from the pin numbers, returning an error if no pinout could be found.

Not yet properly tested. However, while doing some initial testing, I found that there was an inconsistency in the Nano 33 IoT pinout for SPI0: the MISO pin is `A6` or `PA09` which has SERCOM0/PAD1 while the `DIpad` was set to `sercomRXPad0` - thus SERCOM0/PAD0 (pin PA08). I don't know which of the two it is supposed to be and whether I didn't miss anything but I'm surprised this worked. @deadprogram can you take a look?

Also, are there espat examples for the Nano 33 IoT (or other SPI tests) that should work with little or no modification?

This is just the first step, my goal is to convert UART and I2C in the same way. And also see whether this is possible on the SAMD51.

### CODE REVIEW

**Comment by deadprogram** [comment]
The espat driver does not use SPI interface it only uses UART.

**Comment by deadprogram** [comment]
The easiest way to test is probably using a small APA102 strip that is small enough to power from the onboard 3.3V.

I was not able to get this branch working on the ItsyBitsy-m0 using the drivers repo's APA102 example.

File: src/machine/machine_atsamd21.go [hunk 1]
```diff
@@ -103,6 +103,117 @@ const (
...
+		if (upper>>1)-1 == sercom {
+			pinMode = PinSERCOM
+			pad |= uint32((upper & 1) << 1)
+			ok = true
+		}
+	}
+	if lower != 0 {
+		// SERCOM-ALT
+		if (lower>>1)-1 == sercom {
+			pinMode = PinSERCOM
```
Comment (deadprogram) [inline src/machine/machine_atsamd21.go]: This line should be:

```go
pinMode = PinSERCOMAlt
```

**Comment by deadprogram** [comment]
Once I applied the change in the code mentioned above, the APA102 example worked correctly on both the ItsyBitsy-M0 as well as the Arduino Nano33 IoT.

**Comment by aykevl** [comment]
Great, thank you for the testing and finding that bug! I will also test it myself a bit more.

**Comment by aykevl** [comment]
Fixed the bug, so this is (again) ready for testing.

**Comment by deadprogram** [comment]
Confirmed the fix, so now merging. Very nice work @aykevl thanks for the cleanup.

**Pull request merged**